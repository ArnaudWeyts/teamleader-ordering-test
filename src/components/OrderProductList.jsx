import React, { Component } from 'react';

import { Card, Button, InputNumber } from 'antd';

import Product from './Product';
import { formatToPrice } from '../helpers';

class OrderProductList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      removeQuantity: 1
    };

    this.handleInputNumberChange = this.handleInputNumberChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleInputNumberChange(number) {
    this.setState({
      removeQuantity: number
    });
  }

  handleSubmit(e, id) {
    e.preventDefault();
    const { removeProduct } = this.props;
    const { removeQuantity } = this.state;
    removeProduct(id, removeQuantity);
  }

  render() {
    const { items } = this.props;
    const { removeQuantity } = this.state;

    if (!items) {
      return <div>Loading...</div>;
    }

    return (
      <div style={{ maxHeight: '70vh', overflowY: 'scroll' }}>
        {items.map(item => (
          <Card key={item['product-id']}>
            <Product product={item.product} />
            <h3>Quantity: {item.quantity}</h3>
            <h3>Subtotal: {formatToPrice(item.total)}</h3>
            <form onSubmit={e => this.handleSubmit(e, item['product-id'])}>
              <InputNumber
                name="removeQuantity"
                min={1}
                max={+item.quantity}
                value={removeQuantity}
                onChange={this.handleInputNumberChange}
              />
              <Button type="default" htmlType="submit">
                Remove product(s)
              </Button>
            </form>
          </Card>
        ))}
      </div>
    );
  }
}

export default OrderProductList;
