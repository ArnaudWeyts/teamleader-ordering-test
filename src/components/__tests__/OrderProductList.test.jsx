import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import OrderProductList from '../OrderProductList';

describe('OrderProductList tests', () => {
  const items = [
    {
      'product-id': 'A101',
      quantity: '2',
      'unit-price': 9.75,
      total: 19.5,
      product: {
        id: 'A101',
        description: 'Screwdriver',
        category: '1',
        price: 9.75
      }
    },
    {
      'product-id': 'A102',
      quantity: 1,
      'unit-price': 49.5,
      total: 49.5,
      product: {
        id: 'A102',
        description: 'Electric screwdriver',
        category: '1',
        price: 49.5
      }
    }
  ];

  it('renders without crashing', () => {
    shallow(<OrderProductList items={items} />);
  });

  it('renders OrderProductList', () => {
    const component = renderer.create(<OrderProductList items={items} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
