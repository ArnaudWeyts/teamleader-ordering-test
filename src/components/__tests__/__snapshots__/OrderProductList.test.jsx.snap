// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderProductList tests renders OrderProductList 1`] = `
<div
  style={
    Object {
      "maxHeight": "70vh",
      "overflowY": "scroll",
    }
  }
>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div>
        <p>
          Description: 
          Screwdriver
        </p>
        <p>
          Category: 
          1
        </p>
        <p>
          Price: 
          €9.75
        </p>
      </div>
      <h3>
        Quantity: 
        2
      </h3>
      <h3>
        Subtotal: 
        €19.50
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="ant-input-number"
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onMouseOut={undefined}
          onMouseOver={undefined}
          style={Object {}}
        >
          <div
            className="ant-input-number-handler-wrap"
          >
            <span
              aria-disabled={false}
              aria-label="Increase Value"
              className="ant-input-number-handler ant-input-number-handler-up "
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              unselectable="unselectable"
            >
              <i
                className="anticon anticon-up ant-input-number-handler-up-inner"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="up"
                  fill="currentColor"
                  height="1em"
                  onClick={undefined}
                  style={undefined}
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                  />
                </svg>
              </i>
            </span>
            <span
              aria-disabled={true}
              aria-label="Decrease Value"
              className="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              role="button"
              unselectable="unselectable"
            >
              <i
                className="anticon anticon-down ant-input-number-handler-down-inner"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="down"
                  fill="currentColor"
                  height="1em"
                  onClick={undefined}
                  style={undefined}
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </i>
            </span>
          </div>
          <div
            aria-valuemax={2}
            aria-valuemin={1}
            aria-valuenow={1}
            className="ant-input-number-input-wrap"
            role="spinbutton"
          >
            <input
              autoComplete="off"
              autoFocus={undefined}
              className="ant-input-number-input"
              disabled={undefined}
              id={undefined}
              max={2}
              maxLength={undefined}
              min={1}
              name="removeQuantity"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={undefined}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseUp={[Function]}
              pattern={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              step={1}
              tabIndex={undefined}
              type={undefined}
              value="1"
            />
          </div>
        </div>
        <button
          className="ant-btn ant-btn-default"
          onClick={[Function]}
          type="submit"
        >
          <span>
            Remove product(s)
          </span>
        </button>
      </form>
    </div>
  </div>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div>
        <p>
          Description: 
          Electric screwdriver
        </p>
        <p>
          Category: 
          1
        </p>
        <p>
          Price: 
          €49.50
        </p>
      </div>
      <h3>
        Quantity: 
        1
      </h3>
      <h3>
        Subtotal: 
        €49.50
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="ant-input-number"
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onMouseOut={undefined}
          onMouseOver={undefined}
          style={Object {}}
        >
          <div
            className="ant-input-number-handler-wrap"
          >
            <span
              aria-disabled={true}
              aria-label="Increase Value"
              className="ant-input-number-handler ant-input-number-handler-up ant-input-number-handler-up-disabled"
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              role="button"
              unselectable="unselectable"
            >
              <i
                className="anticon anticon-up ant-input-number-handler-up-inner"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="up"
                  fill="currentColor"
                  height="1em"
                  onClick={undefined}
                  style={undefined}
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                  />
                </svg>
              </i>
            </span>
            <span
              aria-disabled={true}
              aria-label="Decrease Value"
              className="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              role="button"
              unselectable="unselectable"
            >
              <i
                className="anticon anticon-down ant-input-number-handler-down-inner"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="down"
                  fill="currentColor"
                  height="1em"
                  onClick={undefined}
                  style={undefined}
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </i>
            </span>
          </div>
          <div
            aria-valuemax={1}
            aria-valuemin={1}
            aria-valuenow={1}
            className="ant-input-number-input-wrap"
            role="spinbutton"
          >
            <input
              autoComplete="off"
              autoFocus={undefined}
              className="ant-input-number-input"
              disabled={undefined}
              id={undefined}
              max={1}
              maxLength={undefined}
              min={1}
              name="removeQuantity"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={undefined}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseUp={[Function]}
              pattern={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              step={1}
              tabIndex={undefined}
              type={undefined}
              value="1"
            />
          </div>
        </div>
        <button
          className="ant-btn ant-btn-default"
          onClick={[Function]}
          type="submit"
        >
          <span>
            Remove product(s)
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
